plugins/plugin-chart-echarts/src/Timeseries/EchartsTimeseries.tsx

```diff
@@ -71,30 +119,83 @@ export default function EchartsTimeseries({
 
   const eventHandlers: EventHandlers = {
     click: props => {
-      const { seriesName: name, value } = props;
-      const xValue = value[0].getTime?.() || value[0];
-      console.log(props, name, xValue);
-      const values = Object.values(selectedValues);
-      if (values.includes(name)) {
-        handleChange(values.filter(v => v !== name));
-      } else {
-        handleChange([name]);
+      if (clickTimer.current) {
+        clearTimeout(clickTimer.current);
       }
+      // Ensure that double-click events do not trigger single click event. So we put it in the timer.
+      clickTimer.current = setTimeout(() => {
+        const { seriesName: name } = props;
+        const values = Object.values(selectedValues);
+        if (values.includes(name)) {
+          handleChange(values.filter(v => v !== name));
+        } else {
+          handleChange([name]);
+        }
+      }, TIMER_DURATION);
     },
     mousemove: params => {
       currentSeries.name = params.seriesName;
     },
     mouseout: () => {
       currentSeries.name = '';
     },
+    legendselectchanged: payload => {
+      const currentTime = Date.now();
+      // TIMER_DURATION is the interval between two legendselectchanged event
+      if (
+        currentTime - lastTimeRef.current < TIMER_DURATION &&
+        lastSelectedLegend.current === payload.name
+      ) {
+        // execute dbclick
+        handleDoubleClickChange(payload.name);
+      } else {
+        lastTimeRef.current = currentTime;
+        // remember last selected legend
+        lastSelectedLegend.current = payload.name;
+      }
+      // if all legend is unselected, we keep all selected
+      if (Object.values(payload.selected).every(i => !i)) {
+        handleDoubleClickChange();
+      }
+    },
+  };
+
+  const zrEventHandlers: EventHandlers = {
```

Comment:
Here we use zrender event (https://echarts.apache.org/handbook/en/concepts/event/#listen-to-events-on-the-blank-area) instead of echart event, because we need to click on non-series areas to filter the series, such as stacked areas.